tools/github-team-user-store/GitHubTeamUserStore/GitHubTeamUserStore/GitHubEventClient.cs

```diff
@@ -0,0 +1,199 @@
+
+        /// <summary>
+        /// Given an Octokit.Team, call to get the team members. Note: GitHub's GetTeamMembers API gets all of the Users
+        /// for the team which includes all the members of child teams.
+        /// </summary>
+        /// <param name="team">Octokit.Team, the team whose members to retrieve.</param>
+        /// <returns>IReadOnlyList of Octokit.Users</returns>
+        /// <exception cref="ApplicationException">Thrown if GetAllMembers fails after all retries have been exhausted.</exception>
+        public async Task<IReadOnlyList<User>> GetTeamMembers(Team team)
+        {
+            ApiOptions apiOptions = new ApiOptions();
+            apiOptions.PageSize = 100;
+
+            // For the cases where exceptions/retries fail and an empty ReadOnlyList needs to be returned
+            List<User> emptyUserList = new List<User>();
+
+            int tryNumber = 0;
+            while (tryNumber < _numRetries)
+            {
+                tryNumber++;
+                try
+                {
+                    return await _gitHubClient.Organization.Team.GetAllMembers(team.Id, apiOptions);
+                }
+                // This is what gets thrown if we try and get a userList for certain special teams on GitHub. 
+                // None of these teams are used directly in anything and neither team is a child team of 
+                // azure-sdk-write. If a ForbiddenException is encountered, then report it and return an
+                // empty list.
+                catch (Octokit.ForbiddenException forbiddenEx)
+                {
+                    Console.WriteLine($"{team.Name} cannot be retrieved using a GitHub PAT.");
+                    Console.WriteLine(forbiddenEx.Message);
+                    return emptyUserList.AsReadOnly();
+                }
+                // The only time we should get down here is if there's an exception caused by a network hiccup.
+                // Sleep for a second and try again.
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.Message);
+                    Console.WriteLine($"delaying {_delayTimeInMs} and retrying");
+                    await Task.Delay(_delayTimeInMs);
+                }
+            }
+            throw new ApplicationException($"Unable to get members for team {team.Name}. See above exception(s)");
+        }
+
+        /// <summary>
+        /// Given an Octokit.Team, call to get all child teams.
+        /// </summary>
+        /// <param name="team">Octokit.Team, the team whose child teams to retrieve.</param>
+        /// <returns>IReadOnlyList of Octkit.Team</returns>
+        /// <exception cref="ApplicationException">Thrown if GetAllChildTeams fails after all retries have been exhausted</exception>
+        public async Task<IReadOnlyList<Team>> GetAllChildTeams(Team team)
+        {
+            ApiOptions apiOptions = new ApiOptions();
+            apiOptions.PageSize = 100;
```

Comment:
Make this `100` into class member? Same for all such inline magical constants.

Reply:
Made the API options a static variable on the class with the default, from a const, MaxPageSize.